src/resources.rs

```diff
@@ -4,16 +4,45 @@ use std::collections::hash_map::{Entry, HashMap};
 use std::hash::{BuildHasherDefault, Hasher};
 use std::ops::{Deref, DerefMut};
 
+/// A type map for holding resources.
+///
+/// In an ECS, resources are "global" singular objects that may be accessed by systems
```

Comment:
I think the comparison with entities is not too helpful for a first entry. Maybe just "resources replace global variables and my be accessed by systems that know their type"?